src/pybind/mgr/poolsets/module.py

```diff
@@ -0,0 +1,1408 @@
+import errno
+import threading
+import json
+import uuid
+from collections import defaultdict
+
+from mgr_module import MgrModule, CommandResult
+
+"""
+Some terminology is made up for the purposes of this module:
+
+ - "raw pgs": pg count after applying replication, i.e. the real resource
+              consumption of a pool.
+ - "grow/shrink" - increase/decrease the pg_num in a pool
+ - "crush subtree" - non-overlapping domains in crush hierarchy: used as
+                     units of resource management.
+"""
+
+
+# We will aim to have this many PGs per OSD
+TARGET_PGS_PER_OSD = 100
```

Comment:
We have a mon_max_pg_per_osd.  Let's add a mon_target_pg_per_osd option?

(We could also s/pg/pgs/ in those)